wrappers/services_test.go

```diff
@@ -3174,12 +3217,69 @@ apps:
 	s.sysdLog = nil
 	c.Assert(wrappers.RestartServices(info.Services(), nil, progress.Null, s.perfTimings), IsNil)
 	c.Check(s.sysdLog, DeepEquals, [][]string{
+		{"show", "--property=Id,ActiveState,UnitFileState,Type", srvFile},
 		{"stop", srvFile},
 		{"show", "--property=ActiveState", srvFile},
 		{"start", srvFile},
 	})
 }
 
+func (s *servicesTestSuite) TestRestartInDifferentStates(c *C) {
+	const manyServicesYaml = `name: test-snap
+version: 1.0
+apps:
+  svc1:
+    command: bin/foo
+    daemon: simple
+  svc2:
+    command: bin/foo
+    daemon: simple
+  svc3:
+    command: bin/foo
+    daemon: simple
+  svc4:
+    command: bin/foo
+    daemon: simple
+`
+	srvFile1 := "snap.test-snap.svc1.service"
+	srvFile2 := "snap.test-snap.svc2.service"
+	srvFile3 := "snap.test-snap.svc3.service"
+	srvFile4 := "snap.test-snap.svc4.service"
+
+	info := snaptest.MockSnap(c, manyServicesYaml, &snap.SideInfo{Revision: snap.R(1)})
+
+	r := systemd.MockSystemctl(func(cmd ...string) ([]byte, error) {
+		s.sysdLog = append(s.sysdLog, cmd)
+		states := map[string]serviceState{
+			srvFile1: {"active", "enabled"},
+			srvFile2: {"inactive", "enabled"},
+			srvFile3: {"active", "disabled"},
+			srvFile4: {"inactive", "disabled"},
+		}
+		if out := mockSystemCtlShow(cmd, states); out != nil {
+			return out, nil
+		}
+		return []byte("ActiveState=inactive\n"), nil
+	})
+	defer r()
+
+	err := wrappers.AddSnapServices(info, nil, progress.Null)
+	c.Assert(err, IsNil)
+
+	s.sysdLog = nil
+	c.Assert(wrappers.RestartServices(info.Services(), nil, progress.Null, s.perfTimings), IsNil)
+	c.Check(s.sysdLog, DeepEquals, [][]string{
+		{"show", "--property=Id,ActiveState,UnitFileState,Type",
```

Comment:
This fails for me randomly right now, it seems the order is not always preserved. Not sure if this is an actual issue or just an issue with this test, i.e. if it's enough to use something less strong than "DeepEquals" (like testutil.Contains) etc.

Reply:
Just to be clear:
```
cd wrappers
go test -check.f TestRestartInDifferentStates -count 50
```
will fail for me with essential different ordering in the systemctl show ordering.

Reply:
hmm yeah I can't tell just by looking at the code where the different ordering is coming from, I don't see anything obvious like using `range` with a map ...

Reply:
Should be fixed now :-)